x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/ToGeohash.java

```diff
@@ -16,53 +16,56 @@
 import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
 import org.elasticsearch.xpack.esql.core.tree.Source;
 import org.elasticsearch.xpack.esql.core.type.DataType;
-import org.elasticsearch.xpack.esql.evaluator.mapper.EvaluatorMapper;
 import org.elasticsearch.xpack.esql.expression.function.Example;
 import org.elasticsearch.xpack.esql.expression.function.FunctionAppliesTo;
 import org.elasticsearch.xpack.esql.expression.function.FunctionAppliesToLifecycle;
 import org.elasticsearch.xpack.esql.expression.function.FunctionInfo;
 import org.elasticsearch.xpack.esql.expression.function.Param;
-import org.elasticsearch.xpack.esql.expression.function.scalar.convert.AbstractConvertFunction;
 
 import java.io.IOException;
 import java.util.List;
 import java.util.Map;
 
+import static org.elasticsearch.xpack.esql.core.type.DataType.GEOHASH;
 import static org.elasticsearch.xpack.esql.core.type.DataType.KEYWORD;
 import static org.elasticsearch.xpack.esql.core.type.DataType.LONG;
+import static org.elasticsearch.xpack.esql.core.type.DataType.TEXT;
 
-public class StGeohashToLong extends AbstractConvertFunction implements EvaluatorMapper {
+public class ToGeohash extends AbstractConvertFunction {
     public static final NamedWriteableRegistry.Entry ENTRY = new NamedWriteableRegistry.Entry(
         Expression.class,
-        "StGeohashToLong",
-        StGeohashToLong::new
+        "ToGeohash",
+        ToGeohash::new
     );
 
     private static final Map<DataType, BuildFactory> EVALUATORS = Map.ofEntries(
         Map.entry(LONG, (source, fieldEval) -> fieldEval),
-        Map.entry(KEYWORD, StGeohashToLongFromStringEvaluator.Factory::new)
+        Map.entry(KEYWORD, ToGeohashFromStringEvaluator.Factory::new),
+        Map.entry(TEXT, ToGeohashFromStringEvaluator.Factory::new)
```

Comment:
How is converting from text reasonable? Wouldn't text normally be tokenized, and thus could be substantially different from the original source?

Reply:
Thanks for the context, makes sense!